internal/cluster/clusters.go

```diff
@@ -100,44 +114,23 @@ func (e *clusterModelNotFoundError) NotFound() bool {
 }
 
 // findOneBy returns a cluster instance for an organization by cluster name.
-func (c *Clusters) findOneBy(organizationID uint, field string, criteria interface{}) (*model.ClusterModel, error) {
-	cluster := model.ClusterModel{
-		OrganizationId: organizationID,
-	}
-
-	switch field {
-	case "id":
-		id, ok := criteria.(uint)
-		if !ok {
-			return nil, errors.New("criteria is not a valid uint value for id")
-		}
-
-		cluster.ID = id
-
-	case "name":
-		name, ok := criteria.(string)
-		if !ok {
-			return nil, errors.New("criteria is not a valid string value for name")
-		}
-
-		cluster.Name = name
-	}
-
-	err := c.db.Where(cluster).Preload("ScaleOptions").First(&cluster).Error
+func (c *Clusters) findOneBy(cluster model.ClusterModel) (*model.ClusterModel, error) {
```

Comment:
i would trigger an error somewhere if all IDs are zero (default).

Reply:
(also take into consideration that iirc there were some calls with orgid=0 and clusterid!=0 that take advantage of that clusterid is unique in itself)